google-cloud-spanner/test/google/cloud/spanner/client/resource_based_routing_test.rb

```diff
@@ -0,0 +1,137 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require "helper"
+
+describe Google::Cloud::Spanner::Client, :resource_based_routing, :mock_spanner do
+  let(:instance_id) { "my-instance-id" }
+  let(:database_id) { "my-database-id" }
+  let(:session_id) { "session123" }
+  let(:session_grpc) { Google::Spanner::V1::Session.new name: session_path(instance_id, database_id, session_id) }
+  let(:session) { Google::Cloud::Spanner::Session.from_grpc session_grpc, spanner.service }
+  let(:pool_opts) { { min: 0, max: 4 } }
+
+  before do
+    session.instance_variable_set :@last_updated_at, Time.now
+    ENV.delete "GOOGLE_CLOUD_ENABLE_RESOURCE_BASED_ROUTING"
+  end
+
+  after do
+    ENV.delete "GOOGLE_CLOUD_ENABLE_RESOURCE_BASED_ROUTING"
+  end
+
+
+  it "sets service with default host" do
+    client = Google::Cloud::Spanner::Client.new \
+      spanner, instance_id, database_id, pool_opts: { min: 0, max: 4 }
+
+    client.service.host.must_equal \
+      Google::Cloud::Spanner::V1::SpannerClient::SERVICE_ADDRESS
+    client.resource_based_routing_enabled?.must_equal false
+  end
+
+  it "sets service host to default service uri if resource based routing disabled" do
+    client = Google::Cloud::Spanner::Client.new \
+      spanner, instance_id, database_id, pool_opts: { min: 0, max: 4 },
+      enable_resource_based_routing: false
+
+    client.service.host.must_equal \
+      Google::Cloud::Spanner::V1::SpannerClient::SERVICE_ADDRESS
+    client.resource_based_routing_enabled?.must_equal false
+  end
+
+  it "set service host to instance endpoint url if resource based routing enabled" do
+    get_res = Google::Spanner::Admin::Instance::V1::Instance.new \
+      name: instance_path(instance_id), endpoint_uris: ["test.host.com"]
```

Comment:
Return multiple endpoints and verify that the first endpoint is used by the client. This will be the standard behavior from the backend.

Reply:
Added multiple endpoints.